x-pack/plugins/ml/public/application/data_frame_analytics/pages/analytics_management/components/analytics_list/action_delete.test.tsx

```diff
@@ -10,14 +10,24 @@ import { render } from '@testing-library/react';
 import * as CheckPrivilige from '../../../../../capabilities/check_capabilities';
 
 import { DeleteAction } from './action_delete';
-
+import { coreMock as mockCoreServices } from '../../../../../../../../../../src/core/public/mocks';
 import mockAnalyticsListItem from './__mocks__/analytics_list_item.json';
 
 jest.mock('../../../../../capabilities/check_capabilities', () => ({
   checkPermission: jest.fn(() => false),
   createPermissionFailureMessage: jest.fn(),
 }));
 
+jest.mock('../../../../../../application/util/dependency_cache', () => ({
+  getToastNotifications: () => ({ addSuccess: jest.fn(), addDanger: jest.fn() }),
+}));
+
+jest.mock('../../../../../contexts/kibana', () => ({
+  useMlKibana: () => ({
+    services: mockCoreServices.createStart(),
+  }),
+}));
+
 describe('DeleteAction', () => {
```

Comment:
could you please add basic unit tests for asserting index pattern check and user permissions?

Reply:
Done in https://github.com/elastic/kibana/pull/66934/files/17b5fdfe7b7c4413b31dabcfee42abb0614eaa63..807196400c4ef7164e5a44c14075d59d4e3fc218